### PR TITLE
Fixed cassandra db QueryTimeout issue

### PR DESCRIPTION
**Pull request by arvindgh**
Added code to get QueryTimeout value in TNS and Added code to check dbtype before creating connectionString

### CODE REVIEW

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/GingerCore/Environments/Database.cs  3
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/GingerCore/Environments/Database.cs  3
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/GingerCore/Environments/Database.cs  3
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/GingerCore/Environments/Database.cs  3
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

File: Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs [hunk 1]
```diff
@@ -251,11 +252,11 @@ private string GetQueryParamater(string inputSQL,string param)
         }
         public override void PerformDBAction()
         {
-            if (!Connect())
-            {
-                Act.Error = "Failed to connect to Mongo DB";
-                return;
-            }
+            //if (!Connect())
+            //{
```
Comment (ambarkhan) [inline Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs:256]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
- Ginger/GingerCore/Environments/Database.cs  3
- Ginger/GingerCore/NoSqlBase/NoSqlBase.cs  2
         

Complexity decreasing per file
==============================
+ Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs  -1
+ Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

File: Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs [hunk 2]
```diff
@@ -136,11 +136,11 @@ private string GetBucketName(string inputSQL)
 
         public override void PerformDBAction()
         {
-            if (!Connect())
-            {
-                Act.Error = "Failed to connect to Couchbase DB";
-                return;
-            }
+            //if (!Connect())
+            //{
```
Comment (ambarkhan) [inline Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs:140]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

File: Ginger/GingerCore/Actions/ActDBValidation.cs [hunk 3]
```diff
@@ -265,28 +265,48 @@ public override void Execute()
             }
         }
 
+        NoSqlBase.NoSqlBase NoSqlDriver = null;
         private void HandleNoSQLDBAction()
         {
-            NoSqlBase.NoSqlBase NoSqlDriver = null;
-
-            switch(this.DB.DBType)
+            //NoSqlDriver = null;
```
Comment (ambarkhan) [inline Ginger/GingerCore/Actions/ActDBValidation.cs:271]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

File: Ginger/GingerCore/NoSqlBase/GingerCassandra.cs [hunk 4]
```diff
@@ -413,7 +415,11 @@ public bool IfUseJSON()
 
         public override void PerformDBAction()
         {
-            Connect();
+            //if (!Connect())
+            //{
```
Comment (ambarkhan) [inline Ginger/GingerCore/NoSqlBase/GingerCassandra.cs:419]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- Ginger/GingerCore/NoSqlBase/NoSqlBase.cs  2
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
- Ginger/GingerCore/Environments/Database.cs  3
         

Complexity decreasing per file
==============================
+ Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs  -1
+ Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

File: Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs (hunk 2 above)
Comment (ambarkhan) [inline Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs:140]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

File: Ginger/GingerCore/Actions/ActDBValidation.cs (hunk 3 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Actions/ActDBValidation.cs:271]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

File: Ginger/GingerCore/NoSqlBase/GingerCassandra.cs (hunk 4 above)
Comment (ambarkhan) [inline Ginger/GingerCore/NoSqlBase/GingerCassandra.cs:419]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

File: Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs (hunk 1 above)
Comment (ambarkhan) [inline Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs:256]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- Ginger/GingerCore/Environments/Database.cs  3
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
- Ginger/GingerCore/NoSqlBase/NoSqlBase.cs  2
         

Complexity decreasing per file
==============================
+ Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs  -1
+ Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

File: Ginger/GingerCore/NoSqlBase/GingerCassandra.cs (hunk 4 above)
Comment (ambarkhan) [inline Ginger/GingerCore/NoSqlBase/GingerCassandra.cs:419]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

File: Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs (hunk 1 above)
Comment (ambarkhan) [inline Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs:256]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

File: Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs (hunk 2 above)
Comment (ambarkhan) [inline Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs:140]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
- Ginger/GingerCore/Environments/Database.cs  3
         

Complexity decreasing per file
==============================
+ Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs  -1
+ Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

File: Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs [hunk 5]
```diff
@@ -36,7 +36,7 @@ public class GingerCouchbase : NoSqlBase
         
      
 
-        public bool Connect()
+        public override bool Connect()
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs:39]: why these empty overrides?

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
- Ginger/GingerCore/Environments/Database.cs  3
         

Complexity decreasing per file
==============================
+ Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs  -1
+ Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/GingerCore/Environments/Database.cs  3
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  2
         

Complexity decreasing per file
==============================
+ Ginger/GingerCore/NoSqlBase/GingerMongoDb.cs  -1
+ Ginger/GingerCore/NoSqlBase/GingerCouchbase.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5368638&bid=17594474)

**Pull request merged**